core/src/avm1/globals/error.rs

```diff
@@ -0,0 +1,66 @@
+//! Error object
+
+use crate::avm1::activation::Activation;
+use crate::avm1::error::Error;
+use crate::avm1::property::Attribute::*;
+use crate::avm1::{Object, ScriptObject, TObject, UpdateContext, Value};
+use gc_arena::MutationContext;
+
+pub fn constructor<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let message = match args.get(0).cloned() {
+        Some(Value::String(s)) => s,
+        Some(v) => v.coerce_to_string(activation, context)?.to_string(),
+        _ => String::new(),
+    };
```

Comment:
Looks like this should actually just copy the value in as is:
```
var e = new Error(111);
trace(e.message); // 111
trace(typeof e.message); // number
```
And it only does so if the arg is not undefined:
```
var e = new Error(undefined);
trace(e.message); // Error
trace(e.hasOwnProperty("message")); // false

var e = new Error("TEST");
trace(e.message); // TEST
trace(e.hasOwnProperty("message")); // true
```

Reply:
Ah, I wasn't sure about what the type should've been since I wasn't able to run it - thanks for checking